internal/log_analysis/log_processor/pantherlog/jsoniter.go

```
@@ -129,6 +129,18 @@ func (*resultEncoder) writePantherFields(r *Result, stream *jsoniter.Stream) {
 	stream.WriteObjectField(FieldParseTimeJSON)
 	stream.WriteVal(r.PantherParseTime)
 
+	if r.PantherSourceID != "" {
```

Do we need such check? can these ever be null? 
Better fail early?

---

This is the equivalent of `omitempty`. This part of the code is the customized  `JSON` encoder for a `pantherlog.Result`. It does not care about where this is used (i.e. log processor, CLI, tests). Source ID is only present in the log processor.